Core/Core/Use Cases/GetObservedStudents.swift

```diff
@@ -0,0 +1,57 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2020-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Foundation
+import CoreData
+
+public class GetObservedStudents: CollectionUseCase {
+    public typealias Model = User
+
+    let observerID: String
+
+    public init(observerID: String) {
+        self.observerID = observerID
+    }
+
+    public var cacheKey: String? {
+        return "get-observed-students-\(observerID)"
+    }
+
+    public var scope: Scope {
+        return .where(#keyPath(User.observerID), equals: observerID, orderBy: #keyPath(User.id))
```

Comment:
Potential issue here: Because this is a CollectionUseCase, it's going to delete all Users that don't match this predicate when it gets refreshed. Which means if another screen writes Users (and they don't set `observerID`), refreshing this use case will wipe out those users and that other screen might know that it needs to be refreshed.

This is the problem with mutable shared data. I think we can continue with this for now but it's something we should be thinking about and keeping an eye on.